### DISCUSSION TITLE
"[@types/emscripten]" Emscripten [FS] Module analyzePath is missing from type definitions

### DISCUSSION THREAD

**Discussion by thenick775** [DefinitelyTyped/DefinitelyTyped#66585] [Issues with a @types package] [answered]
In the emscripten typescript implementation "[@types/emscripten]", the `analyzePath` method is missing from the [FS] namespace/type spec.

See: https://emscripten.org/docs/api_reference/Filesystem-API.html#FS.analyzePath

[a comment by an automated account is not shown]

**Reply by thenick775** [reply]
marked, ty

**Comment by thenick775** [comment]
Should be as simple as including:

```
type FSWithAnalyze = {
    analyzePath: (
      path: string,
      dontResolveLastLink?: boolean
    ) => {
      isRoot: boolean;
      exists: boolean;
      error: Error;
      name: string;
      path: string;
      object: FS.FSNode;
      parentExists: boolean;
      parentPath: stringToUTF16;
      parentObject: FS.FSNode;
    };
  };
```

in the type specs for the FS namespace.

Will work on a PR as I'm available!

**Comment by thenick775** [comment, marked as answer]
Should be fixed in #69386, will validate

**Reply by thenick775** [reply]
Confirmed, thank you @mattfysh! 🚀

**Comment by hedwigz** [comment]
@thenick775 `createNode` is also missing
